nixos/lib/test-driver/test_driver/machine.py

```diff
@@ -1003,6 +1004,33 @@ def forward_port(self, host_port: int = 8080, guest_port: int = 80) -> None:
         """
         self.send_monitor_command(f"hostfwd_add tcp::{host_port}-:{guest_port}")
 
+    def running_under_uefi(self) -> bool:
+        """
+        Returns True if the current environment is running under UEFI, False otherwise.
+        This is achieved by inspecting by the existence of /sys/firmware/efi.
+        """
+        rc, _ = self.execute("test -d /sys/firmware/efi")
+        return (rc == 0)
+
+
+    def read_efi_variable_from_sysfs(self, guid: str, variable_name: str) -> EfiVariable | None:
+        """
+        Read an EFI variable located in efivars sysfs if available.
+        Returns None if the EFI variable does not exist.
+        Raises an assertion error if we are not running under an UEFI environment.
+        """
+        assert self.running_under_uefi(), "This machine is not detected under an UEFI environment"
+        rc, raw_attributes_and_value = self.execute(f"cat /sys/firmware/efi/efivars/{variable_name}-{guid}")
```

Comment:
Technically `execute` would return `Tuple[int, str]` which makes `raw_attributes_and_value` a str, and doesn't convert to bytes directly.
What about:
```suggestion
        rc, raw_attributes_and_value = self.execute(f"base64 /sys/firmware/efi/efivars/{variable_name}-{guid}")
```

and later on a conversion back to bytes?
```python
   raw_attributes_and_value = base64.b64decode(raw_attributes_and_value)
```

Reply:
I had this version before, I don't have a strong opinion, are we certain that base64 can encode the whole space for this? In that case, I guess yes :).

Reply:
(I'm also worried that since that transfers over a tty, that would mess up the content, which base64 encoding should safe-guard).

Reply:
> I had this version before, I don't have a strong opinion, are we certain that base64 can encode the whole space for this? In that case, I guess yes :).

That's my understanding of base64.